.gitignore

```diff
@@ -19,3 +19,9 @@ js/javascript.min.js
 # these are easily recreated on a new project
 # with `npm install`
 node_modules/
+
+# ignore composer dependencies
+# these are easily recreated on a new project
+# with `composer install`
+vendor/
+composer.lock
```

Comment:
If you worry about the (future) dependencies, you can choose to install only patch version of that dependency, so anyone using this theme will install the latest patch version